kyuubi-server/src/main/scala/org/apache/kyuubi/engine/spark/SparkBatchProcessBuilder.scala

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.engine.spark
+
+import java.util.UUID
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable.ArrayBuffer
+
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.operation.log.OperationLog
+import org.apache.kyuubi.server.api.v1.BatchRequest
+import org.apache.kyuubi.util.KyuubiHadoopUtils
+
+class SparkBatchProcessBuilder(
+    override val proxyUser: String,
+    override val conf: KyuubiConf,
+    batchRequest: BatchRequest,
+    override val extraEngineLog: Option[OperationLog] = None)
+  extends SparkProcessBuilder(proxyUser, conf, extraEngineLog) {
+  import SparkProcessBuilder._
+
+  private val batchJobTag = batchRequest.conf.get(TAG_KEY).map(_ + ",").getOrElse("") +
```

Comment:
If the user specifies a non-unique tag, then you getApplications by tags at line78 will not be unique. A comma-separated tag will be split into two tags, and the yarnClient api will match the two tags separately